lib/model/model.go

```diff
@@ -1057,9 +1153,40 @@ func (m *model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterCon
 		if !ok {
 			// Shouldn't happen because !cfg.Paused, but might happen
 			// if the folder is about to be unpaused, but not yet.
+			l.Debugln("ccH: no fset", folder.ID)
 			continue
 		}
 
+		ccDeviceRemote, hasDevice := ccDevicesRemote[folder.ID]
+		ccDeviceLocal, hasDeviceLocal := ccDevicesLocal[folder.ID]
+
+		if err := m.ccCheckEncryption(cfg, folderDevice, ccDeviceRemote, ccDeviceLocal, hasDevice, hasDeviceLocal, deviceCfg.Untrusted); err != nil {
+			sameError := false
+			if devs, ok := m.folderEncFailures[folder.ID]; ok {
+				sameError = devs[deviceID] == err
+			} else {
+				m.folderEncFailures[folder.ID] = make(map[protocol.DeviceID]error)
+			}
+			m.folderEncFailures[folder.ID][deviceID] = err
```

Comment:
IT almost feels like we're writing javascript now.
Maps of maps :(

I guess this can't be handled by the folder and the state kept on the folder, we have to keep piling it onto the model?